### PR TITLE
Fix #23435: Large text cursor when editing dynamics text

### LINKED ISSUE #23435: Very tall text cursor when editing text of dynamics
### Issue type

UI bug (incorrect info or interface appearance)

### Bug description

The text cursor is really tall if you double click a dynamic to edit the text

MS4.3.2
Screenshot 2024-07-02 at 3 59 26 PM

Master
Screenshot 2024-07-02 at 3 59 39 PM

### Steps to reproduce

1. Double click a dynamic cursor will be very tall

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/69917893/75b1fe5c-178e-47c3-af7e-f59e3e3448f9

### MuseScore Version

OS: macOS 14.4, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.0-241840302, revision: github-musescore-musescore-f0f2bd8

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Mac OS 14, Windows 11

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mathesoncalum**
Resolves: #23435

### CODE REVIEW

File: src/framework/draw/types/font.cpp [hunk 1]
```diff
@@ -183,7 +183,7 @@ QFont Font::toQFont() const
 
 Font Font::fromQFont(const QFont& qf, Font::Type type)
 {
-    Font f(qf.family(), type);
+    Font f(qf.family().toStdString(), type);
```
Comment (cbjeukendrup) [inline src/framework/draw/types/font.cpp:186]: This converts back and forth between UTF-16 and UTF-8. Instead, use `Font(FontDataKey(qf.family()), …)`, or `Font(String::fromQString(qf.family()), …)` (or `Font(FontDataKey(String::fromQString(qf.family())), … )`)

File: src/framework/draw/types/font.h [hunk 2]
```diff
@@ -33,6 +33,30 @@ namespace muse::draw {
 class Font
 {
 public:
+    struct FontFamily {
+        FontFamily() = default;
+        FontFamily(const char* id)
+            : m_id(String::fromUtf8(id)) {}
```
Comment (cbjeukendrup) [inline src/framework/draw/types/font.h:39]: Maybe make this `const char16_t*` instead, to avoid the String::fromUtf8 conversion

**Comment by mathesoncalum** [comment]
Thanks @cbjeukendrup - fixed!

**Review by igorkorsukov — APPROVED** [review]

**Review by igorkorsukov — APPROVED** [review]

**Comment by zacjansheski** [comment]
Highlighting and pressing space causes the large text cursor to come back.

Deleting makes a tall text box


https://github.com/user-attachments/assets/037c7821-cd8d-45c9-9aa1-dca0376279b9

**Review by cbjeukendrup — APPROVED** [review]

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/23435 FIXED

**Pull request merged**